types/tx.go

```diff
@@ -207,6 +201,14 @@ func (t TxRecordSet) Validate(maxSizeBytes int64, otxs Txs) error {
 	removedCopy := sortedCopy(t.removed)
 	unmodifiedCopy := sortedCopy(t.unmodified)
 
+	var size int64
+	for _, cur := range append(unmodifiedCopy, addedCopy...) {
+		size += int64(len(cur))
+		if size > maxSizeBytes {
+			return fmt.Errorf("transaction data size %d exceeds maximum %d", size, maxSizeBytes)
```

Comment:
I think `size` here might be misleading in the error message, as it we cut the sum short as soon as it exceeds `maxSizeBytes`. I would just leave `maxSizeBytes` in the message